### PR TITLE
Upgrade to wasmtime 10

### PR DESCRIPTION
**Pull request by rylev**
This upgrades `spin-componentize` to wasmtime 10.0. We'll leave this in draft mode until wasmtime 10.0 is officially released and we can decide whether it's the right time to merge this.

### CODE REVIEW

**Comment by rylev** [comment]
wasmtime 10.0.0 has been released, so I'm opening this up for review.

File: adapters/7513464/wasi_snapshot_preview1.command.wasm
Comment (dicej) [inline adapters/7513464/wasi_snapshot_preview1.command.wasm:1]: Why aren't we building the adapter(s) from source anymore?

**Review by dicej — COMMENTED** [review]
Looks good overall.  Just wondering about the adapter binaries.

File: adapters/7513464/wasi_snapshot_preview1.command.wasm
Comment (rylev) [reply adapters/7513464/wasi_snapshot_preview1.command.wasm:1]: Didn't think there was much advantage. The artifact we would get out would be what's checked in here, so why spend needless CPU time building them?

**Pull request merged**